zingo-cli/src/lib.rs

```diff
@@ -519,37 +519,61 @@ fn dispatch_command_or_start_interactive(cli_config: &ConfigTemplate) {
     if cli_config.command.is_none() {
         start_interactive(command_transmitter, resp_receiver);
     } else {
-        // Optionally wait for background sync to finish before executing command
+        // Optionally wait for background sync to finish before executing command.
+        // Retry requesting status until the channel closes or the percentage field
+        // indicates completion.
         if cli_config.sync && cli_config.waitsync {
             use std::{thread, time::Duration};
+            const COMPLETE_THRESHOLD: f64 = 99.999;
+
             loop {
-                // Poll sync task status
+                // Request machine-readable sync status.
                 command_transmitter
-                    .send(("sync".to_string(), vec!["poll".to_string()]))
+                    .send(("sync".to_string(), vec!["status".to_string()]))
                     .unwrap();
+
                 match resp_receiver.recv() {
                     Ok(resp) => {
-                        if resp.starts_with("Error:") {
-                            eprintln!(
-                                "Sync error while waiting: {resp}\nProceeding to execute the command."
-                            );
-                            break;
-                        } else if resp.starts_with("Sync completed succesfully:") {
-                            // Sync finished; proceed
-                            break;
-                        } else if resp == "Sync task has not been launched." {
-                            // Try to launch sync and continue waiting
-                            command_transmitter
-                                .send(("sync".to_string(), vec!["run".to_string()]))
-                                .unwrap();
-                            let _ = resp_receiver.recv();
-                            thread::sleep(Duration::from_millis(500));
-                        } else {
-                            // Not ready yet
-                            thread::sleep(Duration::from_millis(500));
+                        // Parse JSON and inspect the numeric completion field.
+                        match serde_json::from_str::<serde_json::Value>(&resp) {
+                            Ok(json_val) => {
+                                // Extract both percentage fields as f64.
+                                let blocks_pct_opt = json_val
+                                    .get("percentage_total_blocks_scanned")
```

Comment:
Why use a percentage?   

That's just a lossee re-representation of an underyling state, is it not?

Doesn't the percentage shift as a function of the denominator?

If it does not then isn't it just some fixed value...   if it DOES... then isn't it reporting the same percentage for multiple different degrees of completion?!

Please don't use derived statistics when the actual observable underlying quantity is available.

If what we mean is 10 blocks from the tip, or 100 blocks unfinalized..  or whatever then lets use that!